main/main.c

```diff
@@ -19,18 +19,69 @@
 #include <stdio.h>
 #include <stdbool.h>
 #include <signal.h>
+#include <sys/time.h>
+#include <inttypes.h>
 
 #include <rte_eal.h>
 #include <rte_log.h>
 #include <rte_common.h>
 #include <rte_launch.h>
+#include <rte_cycles.h>
 
 #include "gatekeeper_config.h"
 #include "gatekeeper_net.h"
 
 /* Indicates whether the program needs to exit or not. */
 volatile int exiting = false;
 
+/*
+ * These metrics are system dependent, and 
+ * initialized via time_resolution_init() function.
+ */
+uint64_t cycles_per_sec = 0UL;
+uint64_t picosec_per_cycle = 0UL;
+
+/* Obtain the system time resolution. */
+static int
+time_resolution_init(void)
+{
+	int ret;
+	uint64_t tsc_start;
+	struct timeval tv_start;
+	struct timeval tv_now;
+
+	tsc_start = rte_rdtsc();
+	ret = gettimeofday(&tv_start, NULL);
```

Comment:
If you use `clock_getres()` and `clock_gettime()` instead of `gettimeofday()`, you can improve the precision to nanoseconds on systems that are more precise.